src/function/scalar/list/list_sort.cpp

```diff
@@ -112,7 +112,7 @@ static void ListSortFunction(DataChunk &args, ExpressionState &state, Vector &re
 		}
```

Comment:
Looking at Vector::Flatten it seems it already does nothing for a FLAT_VECTOR, so they can be excluded from the check (this comment is about line 110)

Reply:
Is is necessary to physically Flatten Vectors when we can instead use UnifiedVectorFormat? More a question than a comment 😅

Reply:
Yes, I believe that since we do `result.ToUnifiedFormat(count, lists_data);` later the `Flatten` becomes indeed unnecessary. Good catch! Will push a fix.